packages/ui/components/localize/src/date/parseDate.js

```diff
@@ -56,7 +56,7 @@ export function parseDate(dateString) {
   }
 
   const [year, month, day] = parsedString.split('/').map(Number);
-  const parsedDate = new Date(Date.UTC(year, month - 1, day));
+  const parsedDate = new Date(new Date(year, month - 1, day).toLocaleDateString());
```

Comment:
Should we align with the locale defined locally in `localize` here, instead of browser defaults?
I most cases, they are the same, but they could be different on purpose. And it would probably be more consistent with the rest of the code in localize.

So that we have: `new Date(new Date(year, month - 1, day).toLocaleDateString(localize.locale))` ?

Reply:
This has become `const parsedDate = new Date(new Date(year, month - 1, day));`

So we always use the browser locale, for two reasons:
- it's just the most logical for the end user. If it's 9pm, he/she doesn't want to see the date of tomorrow, because the locale is configured in a certain way 
- there's no 1-1 mapping from locale to timezone (a locale/geographical area could have multiple timezones)